components/fxa-client/android/src/main/java/mozilla/appservices/fxa/FirefoxAccount.kt

```diff
@@ -0,0 +1,153 @@
+     */
+    abstract suspend fun getTokenServerEndpointURL(): String?
+
+    /**
+     * Get the pairing URL to navigate to on the Authority side (typically a computer).
+     *
+     * @return The URL to show the pairing user
+     */
+    abstract fun getPairingAuthorityURL(): String
+
+    /**
+     * Returns current FxA Device ID for an authenticated account.
+     *
+     * @return Current device's FxA ID, if available. `null` otherwise.
+     */
+    abstract fun getCurrentDeviceId(): String?
+
+    /**
+     * Returns session token for an authenticated account.
+     *
+     * @return Current account's session token, if available. `null` otherwise.
+     */
+    abstract fun getSessionToken(): String?
+
+    /**
+     * Indicates if account needs to be re-authenticated via [beginAuthentication].
+     * Most common reason for an account to need re-authentication is a password change.
+     *
+     * TODO this may return a false-positive, if we're currently going through a recovery flow.
+     * Prefer to be notified of auth problems via [AccountObserver], which is reliable.
+     *
+     * @return A boolean flag indicating if account needs to be re-authenticated.
+     */
+    abstract fun accountNeedsReauth(): Boolean
+
+    /**
+     * This method should be called when a request made with an OAuth token failed with an
+     * authentication error. It will re-build cached state and perform a connectivity check.
+     *
+     * In time, fxalib will grow a similar method, at which point we'll just relay to it.
+     * See https://github.com/mozilla/application-services/issues/1263
+     *
+     * @param singleScope An oauth scope for which to check authorization state.
+     * @return An optional [Boolean] flag indicating if we're connected, or need to go through
+     * re-authentication. A null result means we were not able to determine state at this time.
+     */
+    abstract suspend fun checkAuthorizationStatus(singleScope: String): Boolean?
+
+    /**
+     * Register for account events
+     */
+    abstract suspend fun registerAccountEventHandler(handler: AccountEventHandler)
+
+    abstract fun stateManager(): StateManager
+
+    abstract fun deviceManager(): DeviceManager
+}
+
+interface AccountEventHandler {
+    fun onAccountEvent(event: AccountEvent)
+}
+
+sealed class AccountEvent  {
+    /** The account's profile was updated */
+    object ProfileUpdated : AccountEvent()
+
+    /** The authentication state of the account changed - eg, the password changed */
+    data class AccountStateChanged(val state: AccountState) : AccountEvent()
+
+    /** The account itself was destroyed */
+    object AccountDestroyed : AccountEvent()
+
+    /** An incoming command from another device */
+    data class DeviceCommandIncoming(val command: IncomingDeviceCommand) : AccountEvent()
+
+    /** Another device connected to the account */
+    data class devicesChanged(val deviceList: DeviceList) : AccountEvent()
+}
```

Comment:
android-components has too many different event handlers, there's [AccountObserver](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/components/concept/sync/src/main/java/mozilla/components/concept/sync/OAuthAccount.kt#L277-L319), [AccountEventsObserver](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/components/concept/sync/src/main/java/mozilla/components/concept/sync/AccountEvent.kt#L10-L41) and [DeviceConstellationObserver](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/components/concept/sync/src/main/java/mozilla/components/concept/sync/Devices.kt#L113-L115).

I think we should just one event type, but make it an enum / sealed class.  Observers match the event type and ignore the ones they don't care about.